### ISSUE TITLE
Adding plot support for loss

### ISSUE THREAD

**Issue by nkapila6** [probabl-ai/skore#1815]
### Is your feature request related to a problem? Please describe.

When I was creating #1814, I noticed that something like `report.metrics.log_loss().plot()` won't work as `log_loss()` returns a `float` whereas `roc()`, `precision()`, etc return `matplotlib` plot objects.

In the deep learning world, we look mainly on the loss values to observe training stability and generalization compared to eval metrics (used in other algos w no backprop). It would be great to cache loss values for both train and validation sets.

### Describe the solution you'd like

With `skorch`, this should be straightforward as you have access to the history.
![Image](https://github.com/user-attachments/assets/edf4e1b9-5b47-428e-86db-8368e1314a31)

So, the `MetricsAccessor` can be something like: 
```python
if isinstance(skorch NN object):
 extract net history, wrap in styled matplotlib obj, return obj
else: return float val.
```

skorch doc: https://skorch.readthedocs.io/en/stable/history.html

**Comment by MarieSacksick** [comment]
Hello @nkapila6, thank you for this suggestion!  
We will tackle this when we start to extend towards deep learning. Your issue helps us to decide what we should do first to be deep-learning compliant :) !

**Comment by nkapila6** [comment]
Most welcome Marie.

Also, to add from #1814, even Keras wrapper has a history method that can be leveraged.

https://keras.io/api/utils/sklearn_wrappers/
